hs/WebParsing/TimeTableParser.hs

```diff
@@ -36,38 +39,32 @@ insertDepartment :: MonadIO m => ([T.Text], T.Text) -> ReaderT SqlBackend m ()
 insertDepartment (code, name) = 
     insert_ (Tables.Department code name)
 
--- | List of unwanted words to remove when getting department codes.
-wordsToRemove :: [String]
-wordsToRemove = ["courses",
-                "courses]", 
-                "includes",
-                "and",
-                "-",
-                "[",
-                "]",
-                ""]
+-- | Extracts a list of lists of department names followed by unfiltered department codes.
+splitTagsToLists :: String -> Either ParseError [[String]]
+splitTagsToLists input =
+    let tags = endBy line eol
+        line = sepBy cell (char '[')
+        cell = many (noneOf ",\n")
+        eol = char '\n'
+    in  parse tags "(unknown)" input
+
+-- | Extracts tuples containing the department names and respective codes from raw data.
+extractDeptInfo :: [[[String]]] -> [([T.Text],T.Text)]
```

Comment:
This is at least one nested list level too high. Its signature should be `[[String]] -> ([T.Text], T.Text)`, and you should `map` it below.